src/lib.rs

```diff
@@ -3453,6 +3453,44 @@ pub unsafe trait FromBytes: FromZeros {
     where
         Self: Sized;
 
+    /// Zeroes the bytes of `self` and returns them as a `&mut [u8]`.
+    ///
+    /// This permits accessing the bytes of a type which may not implement
+    /// [`IntoBytes`] by first zeroing any padding bytes which would be unsound
+    /// to expose via [`IntoBytes::as_bytes`].
+    ///
+    /// # Examples
+    ///
+    /// TODO
+    #[inline]
+    fn zero_and_get_bytes(&mut self) -> &mut [u8] {
```

Comment:
TODO: Bikeshed name

Reply:
My thoughts on the name:

- `zero_and_get_bytes` is a good and clear name, though its length is right on that border of being verbose.
- `mut` could be mentioned but is not necessary when the name `zero` implies mutation.
- `fn zero(&mut self)` on `FromZeros` could yield `&[u8]` of all zeros, and that could be given the same name. I struggle to consider a situation where it's practically useful to have access to an all-zeros `&[u8]` with the same address as some `&mut T`, so 🤷🏻‍♀️